pkg/resource/deploy/step_generator.go

```diff
@@ -140,11 +140,22 @@ func (sg *stepGenerator) checkParent(parent resource.URN, resourceType tokens.Ty
 			}
 		} else {
 			// Else try and set it to the root stack
-			for urn := range sg.urns {
-				if urn.Type() == resource.RootStackType {
-					return urn, nil
-				}
-			}
+
+			// TODO: It looks like this currently has some issues with state ordering (see
+			// https://github.com/pulumi/pulumi/issues/10950). Best I can guess is the stack resource is
+			// hitting the step generator and so saving it's URN to sg.urns and issuing a Create step but not
+			// actually getting to writing it's state to the snapshot. Then in parallel with this something
+			// else is causing a pulumi:providers:pulumi default provider to be created, this picks up the
+			// stack URN from sg.urns and so sets it's parent automatically, but then races the step executor
+			// to write itself to state before the stack resource manages to. Long term we want to ensure
+			// there's always a stack resource present, and so that all resources (except the stack) have a
+			// parent (this will save us some work in each SDK), but for now lets just turn this support off.
+
+			//for urn := range sg.urns {
+			//	if urn.Type() == resource.RootStackType {
+			//		return urn, nil
+			//	}
+			//}
```

Comment:
How will this change a user's existing programs/state? If we merge this, will the URNs of every unparented resource change to no longer have the stack as their parent & will that cause any replaces?

Reply:
Hmm good point, I had been thinking every actual user program would be waiting for and parenting to the stack anyway (the intention was to use this logic to reduce some SDK code but it hasn't been done yet). But that won't apply to default providers, so all default providers might change but then they would of also changed when we added this code and we didn't get any odd reports so I think defaults are special enough that it doesn't matter.

Reply:
> so I think defaults are special enough that it doesn't matter.

Going to pull this down and do a couple quick tests to make sure :)

Reply:
No diff on pulumi up/preview when using 3.46.1 vs 3.46.1-dev.0 on this branch. 👍